modules/dbparser/pdb-file.c

```diff
@@ -76,3 +78,88 @@ pdb_file_detect_version(const gchar *pdbfile, GError **error)
     }
   return result;
 }
+
+static const gchar *
+_get_xsddir_in_build(void)
+{
+#ifdef SYSLOG_NG_ENABLE_DEBUG
+  const gchar *srcdir;
+  static gchar path[256];
+
+  srcdir = getenv("top_srcdir");
+  if (srcdir)
+    {
+      g_snprintf(path, sizeof(path), "%s/doc/xsd", srcdir);
+      return path;
+    }
+#endif
+  return NULL;
+}
+
+static const gchar *
+_get_xsddir_in_production(void)
+{
+  return get_installation_path_for(SYSLOG_NG_PATH_XSDDIR);
+}
+
+static const gchar *
+_get_xsddir(void)
+{
+  return _get_xsddir_in_build() ? : _get_xsddir_in_production();
```

Comment:
This works fine, although I would probably have `#ifdef`'ed the whole function definition above and this condition as well.